stablehlo/transforms/StablehloAggressiveSimplification.cpp

```diff
@@ -245,17 +226,16 @@ static mlir::stablehlo::ComparisonDirection invertDirection(
 
   switch (direction) {
     case ComparisonDirection::EQ:
-      return ComparisonDirection::EQ;
+    case ComparisonDirection::NE:
+      return direction;
     case ComparisonDirection::GE:
       return ComparisonDirection::LE;
-    case ComparisonDirection::LE:
-      return ComparisonDirection::GE;
     case ComparisonDirection::GT:
       return ComparisonDirection::LT;
+    case ComparisonDirection::LE:
+      return ComparisonDirection::GE;
     case ComparisonDirection::LT:
       return ComparisonDirection::GT;
-    case ComparisonDirection::NE:
-      return ComparisonDirection::NE;
   }
 
```

Comment:
Prefer `llvm::report_fatal_error` to `unreachable`, since unreachable may be removed in release builds I believe?

Reply:
See: https://discourse.llvm.org/t/llvm-unreachable-is-widely-misused/60587/21

Reply:
Thanks for sharing the link. Changed to `report_fatal_error`.